semantics/src/errors.rs

```diff
@@ -1,16 +1,131 @@
 //! Semantic errors.
 
+use ansi_term::Color::Red;
+use fe_parser::span::Span;
+
 /// Errors for things that may arise in a valid Fe AST.
 #[derive(Debug, PartialEq)]
-pub enum SemanticError {
+pub enum ErrorKind {
     BreakWithoutLoop,
     ContinueWithoutLoop,
     MissingReturn,
-    NotAnExpression,
     NotSubscriptable,
-    UndefinedValue { value: String },
+    UndefinedValue,
     UnexpectedReturn,
     TypeError,
     CannotMove,
     NotCallable,
 }
+
+#[derive(Debug, PartialEq)]
+pub struct SemanticError {
+    pub kind: ErrorKind,
+    pub context: Vec<Span>,
+}
+
+impl SemanticError {
+    /// Create a new error with kind `BreakWithoutLooping`
+    pub fn break_without_loop() -> Self {
+        SemanticError {
+            kind: ErrorKind::BreakWithoutLoop,
+            context: vec![],
+        }
+    }
+
+    /// Create a new error with kind `ContinueWithoutLoop`
+    pub fn continue_without_loop() -> Self {
```

Comment:
You probably consider these two alternatives but still found the usage too clunky?

1. Write an `impl ErrorKind`

```
impl ErrorKind {

    pub fn to_semantic_error(self) -> SemanticError {
        SemanticError {
            kind: self,
            context: vec![],
        }
    }
}
```

Usage: `ErrorKind::ContinueWithoutLoop.to_semantic_error()`

2. 

```
impl SemanticError {

    pub fn from_error_kind(error_kind: ErrorKind) -> Self {
        SemanticError {
            kind: error_kind,
            context: vec![],
        }
    }
}

...
```

Usage: `SemanticError.from_error_kind(ErrorKind::ContinueWithoutLoop)`